Library/Homebrew/test/cmd/install_spec.rb

```diff
@@ -73,4 +73,27 @@ def install
       .and be_a_success
     expect(HOMEBREW_CELLAR/"testball1/HEAD-d5eb689/foo/test").not_to be_a_file
   end
+
+  it "can install Formulae with an on_linux block" do
+    setup_test_formula "testball1", <<~RUBY
+      version "1.0"
+
+      def install
+        on_macos do
+          # This test is running on Linux so this should not happen
+          raise
+        end
+
+        on_linux do
+          prefix.install Dir["*"]
+        end
+      end
+    RUBY
+
+    expect { brew "install", "testball1" }
```

Comment:
This shouldn't be an integration test but a `formula_spec` test instead.

Reply:
I tried that initially  but I did not find any `install` calls in `formula_spec`. I need to dig further if that is possible there.

Reply:
Look for references to `f.install` and `f.brew { f.install }`